packages/wonder-blocks-grid/components/row.js

```diff
@@ -88,33 +99,72 @@ export default class Row extends React.Component<{
         // that is still visible.
         const filteredContents = [];
         let hasVisibleCell = false;
+        let lastCell;
 
         for (const item of contents) {
             if (
-                !item.type ||
-                !item.props ||
-                !item.type.shouldDisplay ||
-                (typeof item.type.shouldDisplay === "function" &&
-                    item.type.shouldDisplay(item.props, gridSize))
+                typeof item === "object" &&
+                (!item.type ||
+                    !item.props ||
+                    !item.type.shouldDisplay ||
+                    (typeof item.type.shouldDisplay === "function" &&
+                        item.type.shouldDisplay(item.props, gridSize)))
             ) {
-                if (hasVisibleCell) {
+                // Figure out if we need to try and insert a gutter in-between
+                // cells. If the contents are flush then we don't insert any.
+                if (hasVisibleCell && !flush) {
                     filteredContents.push(
                         <Gutter key={`gutter-${filteredContents.length}`} />,
                     );
                 }
 
-                filteredContents.push(item);
+                let newItem: Object = item;
+
+                // If the contents are flush then we insert in new padding to
+                // the cell to implement the first margin and the gutter spacing.
+                if (flush) {
```

Comment:
I feel like `flush` is probably not a good name for what this does.  When I think about flush, I think about the contents lines up with the container.  I see... it's that the contents don't have any extra padding so we're extra padding.  Maybe we should call it `addPaddingToRow` or something like that.

Reply:
Well, in the context of a modal the padding exists - it's just outside of the modal and is invisible :D by putting it inside it becomes visible! If this was done in another context where more background colors were at play it would be a bit more obvious.